### PR TITLE
Tasks 6385485, 6385486 & 4844014: New VaultBase, Client Adapter and blocking Container cmdlets

### PR DESCRIPTION
**Pull request by dragonfly91**
Creating a common VaultBase for both ARSVault and BackupVault.

### CODE REVIEW

File: src/ResourceManager/AzureBackup/Commands.AzureBackup.Test/ScenarioTests/AzureBackupVaultTests.cs [hunk 1]
```diff
@@ -24,6 +24,13 @@ public class AzureBackupVaultTests : AzureBackupTestsBase
         public void AzureBackupVaultScenarioTests()
         {
             this.RunPowerShellTest("Test-AzureBackupVaultScenario");
-        }        
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void AzureBackupVaultRsVaultScenario()
+        {
```
Comment (pragrawa) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup.Test/ScenarioTests/AzureBackupVaultTests.cs]: Rename it to AzureRecoveryServicesVaultScenario

File: src/ResourceManager/AzureBackup/Commands.AzureBackup.Test/ScenarioTests/AzureBackupVaultTests.cs [hunk 2]
```diff
@@ -24,6 +24,13 @@ public class AzureBackupVaultTests : AzureBackupTestsBase
         public void AzureBackupVaultScenarioTests()
         {
             this.RunPowerShellTest("Test-AzureBackupVaultScenario");
-        }        
+        }
+
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void AzureBackupVaultRsVaultScenario()
```
Comment (pragrawa) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup.Test/ScenarioTests/AzureBackupVaultTests.cs]: Rename it to AzureRecoveryServiceVaultScenario

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupVaultCmdletBase.cs [hunk 3]
```diff
@@ -21,14 +22,26 @@ public abstract class AzureBackupVaultCmdletBase : AzureBackupCmdletBase
     {
         [Parameter(Position = 0, Mandatory = true, HelpMessage = AzureBackupCmdletHelpMessage.Vault, ValueFromPipeline = true)]
         [ValidateNotNullOrEmpty]
-        public AzureRMBackupVault Vault { get; set; }
+        public VaultBase Vault { get; set; }
+
+        public AzureRMBackupVault BackupVault { get; set; }
```
Comment (pragrawa) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupVaultCmdletBase.cs]: Rename it to RecoveryServiceVault

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureBackupClientAdapter.cs [hunk 4]
```diff
@@ -49,12 +34,6 @@ public partial class AzureBackupClientAdapter
         private BackupServicesManagementClient azureBackupClient;
```
Comment (sudhakar4171) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureBackupClientAdapter.cs]: These two names are confusing - azureBackupVaultClient and azureBackupClient

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureBackupClientAdapter.cs [hunk 5]
```diff
@@ -49,12 +34,6 @@ public partial class AzureBackupClientAdapter
...
 
         /// <summary>
-        /// Cancellation Token Source
-        /// </summary>
-        private CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
-        private CancellationToken CmdletCancellationToken;
-
```
Comment (sudhakar4171) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureBackupClientAdapter.cs]: If these CancellationTokens are not required, then why we defined them earlier and now deleting?

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureRsClientAdapter.cs [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+        /// <summary>
+        /// Azure Recovery Services Backup client
+        /// </summary>
+        private RecoveryServicesBackupManagementClient azureRsClient;        
+
+        /// <summary>
+        /// Azure Recovery Services Backup client
+        /// </summary>
+        private RecoveryServicesBackupManagementClient AzureRSClient
```
Comment (sudhakar4171) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureRsClientAdapter.cs]: pls use same format everywhere - someplaces Rs, others RS

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupVaultCmdletBase.cs [hunk 7]
```diff
@@ -21,14 +24,30 @@ public abstract class AzureBackupVaultCmdletBase : AzureBackupCmdletBase
     {
         [Parameter(Position = 0, Mandatory = true, HelpMessage = AzureBackupCmdletHelpMessage.Vault, ValueFromPipeline = true)]
         [ValidateNotNullOrEmpty]
-        public AzureRMBackupVault Vault { get; set; }
+        public VaultBase Vault { get; set; }
```
Comment (sudhakar4171) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupVaultCmdletBase.cs]: Did you talk to PM if VaultBase name is OK?

File: src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureBackupClientAdapter.cs [hunk 8]
```diff
@@ -21,24 +21,9 @@
 
 namespace Microsoft.Azure.Commands.AzureBackup.ClientAdapter
 {
-    public partial class AzureBackupClientAdapter
+    public partial class AzureBackupClientAdapter : ClientAdapterBase
```
Comment (sudhakar4171) [inline src/ResourceManager/AzureBackup/Commands.AzureBackup/AzureBackupClientAdapter/AzureBackupClientAdapter.cs]: duplicate code across clientadapter. As discussed try to use templatized way.

**Pull request merged**